client/src/app.js

```diff
@@ -0,0 +1,94 @@
+import Component from "./core/component.js";
+import Header from "./components/Header/header.js"
+import MainPage from "./components/mainPage.js";
+import { initialRoutes } from './router.js';
+import SubPage from "./components/SubPages/subPage.js";
+import FavoritePage from "./components/Favorite/favoritePage.js";
+
+export default class App extends Component{
+
+  template(){
+    return `
+      <Header data-component="Header"></Header>
+      <div id="page"></div>
+      `;
+  }
+
+  mount(){
+    
+    const pageDiv=this.$target.querySelector("#page");
+    initialRoutes(pageDiv);
+
+    const $header=this.$target.querySelector('[data-component="Header"]');
+    const $mainPage=this.$target.querySelector('[data-component="MainPage"]');
+    const $subPage=this.$target.querySelector('[data-component="SubPage"]');
+
+    new Header($header,null);
+    $mainPage && new MainPage($mainPage,this.$props);
+    $subPage && new SubPage($subPage,this.$props);
+
+  }
+
+  setEvent(){
+    window.onload=()=>{
+      this.addEvent('clic k','.route',(e)=>{
+        this.removeHeaderStyle(e);
+      })
+
+      window.addEventListener('hashchange', () => {
+        this.routingPageRandering();
+      })
+
+      window.addEventListener('keydown',(e)=>{
+        if(e.key == 'F5' || e.ctrlKey == true && (e.key == 'r')){
+          e.cancelBubble = true;
+          e.returnValue = false;
+          location.href='/'
+          setTimeout(function(){
+              window.location.reload();
+          }, 1000);
+          return false;
+        }
+      })
+    }
+  }
+
+  routingPageRandering(){
+    const link=window.location.hash;
+    switch(link){
+      case '#Home':
+        const $mainPage=this.$target.querySelector('[data-component="MainPage"]');
+        new MainPage($mainPage,this.$props);
+        break;
+
+      case '#Life':
+      case '#Food':
+      case '#Trip':
+      case '#Culture':
+        const $subPage=this.$target.querySelector('[data-component="SubPage"]');
+        new SubPage($subPage,{props:this.$props,link:link});     
```

Comment:
```suggestion
        const $subPage = this.$target.querySelector('[data-component="SubPage"]');
        new SubPage($subPage, { ...this.$props, link });     
```
이런 방법도 있습니다. 그리고 띄어쓰기 신경써주시면 더 좋을 것 같습니다.